File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.kt
class Solution {
    fun tribonacci(n: Int): Int {
        var a = 0
        var b = 1
        var c = 1
        
        if(n==0) return 0
        
        for(i in 3..n){
            var d = a+b+c
            a = b
            b = c
            c = d
        }
        return c
    }
}

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        int a = 0;
        int b = 1;
        int c = 1;
        
        if(n==0)return 0;
        
        for(int i =3;i<=n;i++){
            int d = a+b+c;
            a = b;
            b = c;
            c = d;
        }
        return c;
    }
};